src/main/java/com/cloudbees/jenkins/support/filter/ContentMapping.java

```diff
@@ -49,32 +48,45 @@
     private static final String ALT_SEPARATOR = " » ";
 
     private final String original;
-    private final Pattern pattern;
     private final String replacement;
     private final int hashCode;
 
-    private ContentMapping(@Nonnull String original, @Nonnull Pattern pattern, @Nonnull String replacement) {
+    private final String[] originals;
+    private final String[] replacements;
+
+    private ContentMapping(@Nonnull String original, @Nonnull String replacement) {
         this.original = original;
-        this.pattern = pattern;
         this.replacement = replacement;
+
+        // add flavors of the original string to replace, avoid add when equals
+        String slashChangedInOriginal = original.replace("/", ALT_SEPARATOR);
+        List<String> originalsList = new ArrayList<>(4);
+        originalsList.add(original);
+        addIfNotExist(originalsList, Functions.escape(original));
+        addIfNotExist(originalsList, slashChangedInOriginal);
+        addIfNotExist(originalsList, Functions.escape(slashChangedInOriginal));
+        originals = originalsList.toArray(new String[0]);
+
+        // create the replacement array with the same length as the resulting originals
+        replacements = new String[originals.length];
+        for(int i = 0; i < replacements.length; i++) {
+            replacements[i] = replacement;
+        }
+
```

Comment:
Add all the flavors we had with RegExp: / -> ALT_SEPARATOR, < -> &lt;, \n -> <br>, ...It was missed in the previous version.